src/ray/raylet/worker_pool.cc

```diff
@@ -190,10 +190,26 @@ Process WorkerPool::StartWorkerProcess(const Language &language,
   Process proc = StartProcess(worker_command_args);
   RAY_LOG(DEBUG) << "Started worker process of " << workers_to_start
                  << " worker(s) with pid " << proc.GetId();
+  AddTimerForStartingWorkerProcess(proc, language);
   state.starting_worker_processes.emplace(proc, workers_to_start);
   return proc;
 }
 
+void WorkerPool::AddTimerForStartingWorkerProcess(const Process &proc, const Language &language) {
+  constexpr static size_t timeout_seconds = 3;
```

Comment:
Make this configurable?

Reply:
3 seconds might be too small, in case the node is temporarily frozen. I think 30 seconds is fine.

Reply:
@Kfstorm I don't think it's necessary to configure this because it's a very internal configuration. It's better to expose fewer configurations to avoid complex usage on the user end. `internal_config` is a user end configuration as well.